### PR TITLE
NIP-13: Proof of Work

### PR DESCRIPTION
**Pull request by jb55**
Proof of work for nostr events! I also fixed some other minor readme stuff.

### CODE REVIEW

**Comment by jb55** [comment]
Harding made a good point about [committing to difficulty](https://twitter.com/hrdng/status/1522291030690729985)

> Alice won't look at a Nostr message unless it has 30 bits PoW.  Bob will look at 29 bits.  Without commitments, Carol can target 29 bits for her message to always be seen by Bob and 50% of the time be seen by Alice.  With targets, Alice gets her wish and never sees Carol's msgs.

Will update the nip with this recommendation. Perhaps it can be a third entry in the nonce tag?

**Review by Cameri — APPROVED** [review]

**Comment by Cameri** [comment]
@jb55 I like Hardings' example. We should probably include it.

**Comment by jb55** [comment]
how's this:

> The third entry to the nonce tag SHOULD contain the target difficulty. This allows clients to protect against situations where bulk spammers targeting a lower difficulty get lucky and match a higher difficulty. For example, if you require 40 bits to reply to your thread and see a committed target of 30, you can safely reject it even if the note has 40 bits difficulty. Without a committed target difficulty you could not reject it. Committing to a target difficulty is something all honest miners should be ok with, and clients MAY reject a note matching a target difficulty if it is missing a difficulty commitment.

File: 13.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+```
+$ echo '["REQ", "subid", {"ids": ["000000000"]}]'  | websocat wss://some-relay.com | jq -c '.[2]'
+{"id":"000000000121637feeb68a06c8fa7abd25774bdedfa9b6ef648386fb3b70c387", ...}
+```
+
+Delegated Proof of Work
+-----------------------
+
+Since the `NIP-01` note id does not commit to any signature, PoW can be outsourced to PoW providers, perhaps for a fee. This provides a way for clients to get their messages out to PoW restricted relays without having to do any work themselves, which is useful for energy constrained devices like on mobile
```
Comment (ok300) [inline 13.md:93]: This section here was inspiration for https://oak-node.net/pow :)

Comment (jb55) [reply 13.md:93]: this is cool 👍 amazing work

**Pull request merged**